src/asprofPrivate.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ASPROF_PRIVATE_H
+#define _ASPROF_PRIVATE_H
+
+#include "asprof.h"
+
+// Private functions for the native API.
```

Comment:
This "private" header looks somewhat artificial design-wise. There is no real division into public/private headers in the sources.
It's true that `asprof.h` lists publicly exported functions, and it's OK for other parts of the profiler to use these functions, but if some source needs to call an internal function, then this function has likely nothing to do with `asprof*` API.
In your particular case, `FlightRecorder` is the only caller of `asprofIncrementThreadLocalSampleCounter`, so why can't this function be a part of `FlightRecorder` then? Or perhaps a member of `Profiler`? When we have more thread local data in the future, it may be reasonable to extract these functions to a separate `threadLocal.{h,cpp}` or `tls.{h,cpp}`.